woocommerce/woocommerce
# Remove double checking for woocommerce_tax_display_cart filter

Duplicate of #36885. For original PR:

### All Submissions:






### Changes proposed in this Pull Request:



Closes #36794

This removes the duplicate checking of `woocommerce_tax_display_cart` filter, because of which it does not work correctly and does not honour the `tax_display` param passed to `get_discount_to_display` function.






### How to test the changes in this Pull Request:

It's not possible to test this on a vanilla WC installation, so we will test it using shell.

1.  Update the setting so that price excluding tax is enabled, from WC > Settings > Tax > Prices entered with taxes > No, I will enter prices exclusive of tax.
2. Create a new product, with price set to 10. Note it's ID, let's say its PID.
3. Create a new Coupon, with a percentage discount of 10. Note the coupon code, let's say its COUPONCODE.
4. Set the tax rate to 20%.
5. Open a WC shell and run the following command

```php
update_option( 'woocommerce_tax_display_cart', 'incl' ); //changing to non-default value which exposes the bug in the first palce.
$order = wc_create_order();
$product = wc_get_product( PID );

$order->add_product( $product );
$order->apply_coupon( 'COUPONCODE' );
$order->calculate_totals();
$order->save();

$order->get_discount_to_display( 'excl' ); // should be 1
$order->get_discount_to_display( 'incl' ); // should be 1.2
```






### Other information:




### FOR PR REVIEWER ONLY:
